src/main/java/org/opensearch/neuralsearch/query/NeuralSparseQueryBuilder.java

```diff
@@ -257,15 +267,24 @@ private static void validateQueryTokens(Map<String, Float> queryTokens) {
     protected boolean doEquals(NeuralSparseQueryBuilder obj) {
         if (this == obj) return true;
         if (obj == null || getClass() != obj.getClass()) return false;
+        if (queryTokensSupplier == null && obj.queryTokensSupplier != null) return false;
```

Comment:
May be: if (Objects.isNull(queryTokensSupplier) != Objects.isNull(obj.queryTokensSupplier) return false;

Reply:
I agree this is a more neat style. Since we have the "== null" expression in both main and 2.x, I prefer to create a dedicate PR to fix the code style and do backport. Otherwise we're always creating different PR in main and 2.x.

Of course if you insist we should do the tidy in this PR I'll fix it.

Reply:
Since it is just one place, let's fix this here. Thanks

Reply:
Done.